src/zap.spec.ts

```diff
@@ -0,0 +1,145 @@
+import { PactClient } from "./client";
+import {
+  addLiquidity,
+  algod,
+  createAsset,
+  makeFreshTestBed,
+  signAndSend,
+} from "./testUtils";
+
+describe("zap", () => {
+  it("Calculates all zap params", async () => {
+    const { pool, account } = await makeFreshTestBed({
+      poolType: "CONSTANT_PRODUCT",
+    });
+
+    await addLiquidity(account, pool, 100_000, 100_000);
+    await pool.updateState();
+
+    const zapPrimaryAdd = pool.prepareZap({
+      amount: 10_000,
+      asset: pool.primaryAsset,
+      slippagePct: 2,
+    });
+    expect(zapPrimaryAdd.params).toMatchObject({
+      swapDeposited: 4888n,
+      swapReceived: 4660n,
+      primaryAddLiq: 5112n,
+      secondaryAddLiq: 4646n,
+    });
+
+    const zapSecondaryAdd = pool.prepareZap({
+      amount: 10_000,
+      asset: pool.secondaryAsset,
+      slippagePct: 2,
+    });
+    expect(zapSecondaryAdd.params).toMatchObject({
+      swapDeposited: 4888n,
+      swapReceived: 4660n,
+      primaryAddLiq: 4646n,
+      secondaryAddLiq: 5112n,
+    });
+
+    const { pool: unbalancedPool, account: acc2 } = await makeFreshTestBed({
+      poolType: "CONSTANT_PRODUCT",
+    });
+
+    await addLiquidity(acc2, unbalancedPool, 100_000, 10_000);
```

Comment:
Generally, I put a comment describing how this case differs from the previous one and why it's important to test.